src/app/wallet/wallet.component.ts

```diff
@@ -55,6 +59,106 @@ export class WalletComponent implements OnInit {
     });
   }
 
+
+  // brootle update start
+
+  sortHodlingsToHigh(hodlings: BalanceEntryResponse[]): void {
+    hodlings.sort((a: BalanceEntryResponse, b: BalanceEntryResponse) => {
+      return (
+        this.globalVars.bitcloutNanosYouWouldGetIfYouSold(a.BalanceNanos, a.ProfileEntryResponse.CoinEntry) -
+        this.globalVars.bitcloutNanosYouWouldGetIfYouSold(b.BalanceNanos, b.ProfileEntryResponse.CoinEntry) 
+      );
+    });
+  }  
+
+  toggleSortCoins(){
+    if(this.sortedFromHighToLow){ 
+      this.sortHodlings(this.usersYouPurchased);
+      this.sortHodlings(this.usersYouReceived);        
+      this.sortedFromHighToLow = 0;   
+    } else {
+      this.sortHodlingsToHigh(this.usersYouPurchased);
+      this.sortHodlingsToHigh(this.usersYouReceived);           
+      this.sortedFromHighToLow = 1;    
+    }
+    this.sortedPriceFromHighToLow = -1;
+    this.sortedUsernameFromHighToLow = -1;
+  }
+
+  // sort by coin price
+
+  sortHodlingsPriceToHigh(hodlings: BalanceEntryResponse[]): void {
+
+    hodlings.sort((a: BalanceEntryResponse, b: BalanceEntryResponse) => {
+      return (
+        this.globalVars.bitcloutNanosYouWouldGetIfYouSold(a.ProfileEntryResponse.CoinPriceBitCloutNanos, a.ProfileEntryResponse.CoinEntry) -
+        this.globalVars.bitcloutNanosYouWouldGetIfYouSold(b.ProfileEntryResponse.CoinPriceBitCloutNanos, b.ProfileEntryResponse.CoinEntry) 
+      );
+    });
+  }  
+
+  sortHodlingsPriceToLow(hodlings: BalanceEntryResponse[]): void {
+
+    hodlings.sort((a: BalanceEntryResponse, b: BalanceEntryResponse) => {
+      return (
+        this.globalVars.bitcloutNanosYouWouldGetIfYouSold(b.ProfileEntryResponse.CoinPriceBitCloutNanos, b.ProfileEntryResponse.CoinEntry) -
+        this.globalVars.bitcloutNanosYouWouldGetIfYouSold(a.ProfileEntryResponse.CoinPriceBitCloutNanos, a.ProfileEntryResponse.CoinEntry)
+      );
+    });
+  }    
+
+  toggleSortPrice(){
+    if(this.sortedPriceFromHighToLow){
+      this.sortHodlingsPriceToLow(this.usersYouPurchased);
+      this.sortHodlingsPriceToLow(this.usersYouReceived);         
+      this.sortedPriceFromHighToLow = 0;
+    } else {
+      this.sortHodlingsPriceToHigh(this.usersYouPurchased);
+      this.sortHodlingsPriceToHigh(this.usersYouReceived);          
+      this.sortedPriceFromHighToLow = 1;    
+    }
+    this.sortedFromHighToLow = - 1;
+    this.sortedUsernameFromHighToLow = -1;   
+  }
+
+  // sort by username
+  sortHodlingsUsernameToHigh(hodlings: BalanceEntryResponse[]): void {
+
+    hodlings.sort((a: BalanceEntryResponse, b: BalanceEntryResponse) => {
+      return(
+        a.ProfileEntryResponse.Username.localeCompare(b.ProfileEntryResponse.Username)
+      )
+    });
+
+  }  
+
+  sortHodlingsUsernameToLow(hodlings: BalanceEntryResponse[]): void {
```

Comment:
maybe we just add a param to one of these to indicate that we reverse it.  These are basically the same functions. e.g. `sortHodlingsUsername(hodlings: BalanceEntryResponse, descending: boolean): void`
and if descending, just multiply by -1

Reply:
apply the same logic to the above functions for price and USD value

Reply:
Something like this?

```
  sortHodlingsUsername(hodlings: BalanceEntryResponse[], descending: boolean): void{
    if (descending){
      hodlings.sort((a: BalanceEntryResponse, b: BalanceEntryResponse) => {
        return (
          a.ProfileEntryResponse.Username.localeCompare(b.ProfileEntryResponse.Username)
        )
      });
    }else {
      hodlings.sort((a: BalanceEntryResponse, b: BalanceEntryResponse) => {
        return (
          b.ProfileEntryResponse.Username.localeCompare(a.ProfileEntryResponse.Username)
        )
      });
    }
    this.sortedUsernameFromHighToLow = descending ? -1 : 1;
    this.sortedUSDValueFromHighToLow = 0;
    this.sortedPriceFromHighToLow = 0;
  }

  toggleSortUsername(){
    if (this.sortedUsernameFromHighToLow === 1 || this.sortedUsernameFromHighToLow === 0){     
      this.sortHodlingsUsername(this.usersYouPurchased, true);
      this.sortHodlingsUsername(this.usersYouReceived, true);
    } else {
      this.sortHodlingsUsername(this.usersYouPurchased, false);
      this.sortHodlingsUsername(this.usersYouReceived, false);   
    } 
  }
```

Reply:
```
  sortHodlingsUsername(hodlings: BalanceEntryResponse[], descending: boolean): void {
    this.sortedUsernameFromHighToLow = descending ? -1 : 1;
    this.sortedUSDValueFromHighToLow = 0;
    this.sortedPriceFromHighToLow = 0;
    hodlings.sort((a: BalanceEntryResponse, b: BalanceEntryResponse) => {
        return (
          this.sortedUsernameFromHighToLow * b.ProfileEntryResponse.Username.localeCompare(a.ProfileEntryResponse.Username)
        )
      });
  }

  toggleSortUsername(){
    if (this.sortedUsernameFromHighToLow === 1 || this.sortedUsernameFromHighToLow === 0){     
      this.sortHodlingsUsername(this.usersYouPurchased, true);
      this.sortHodlingsUsername(this.usersYouReceived, true);
    } else {
      this.sortHodlingsUsername(this.usersYouPurchased, false);
      this.sortHodlingsUsername(this.usersYouReceived, false);   
    } 
  }
```

Reply:
I was thinking something like that so we don't have to have two different calls to sort

Reply:
ok, refactored, have a look